packages/rsocket-core/src/RSocketServer.ts

```diff
@@ -22,120 +29,243 @@ export type ServerConfig = {
-                  connection
-                )
-              : undefined;
-            const requester = new RSocketRequester(
-              connection,
-              this.fragmentation?.maxOutboundFragmentSize ?? 0,
-              leaseHandler
-            );
-            const responder = await this.acceptor.accept(
-              {
-                data: frame.data,
-                dataMimeType: frame.dataMimeType,
-                metadata: frame.metadata,
-                metadataMimeType: frame.metadataMimeType,
-                flags: frame.flags,
-                keepAliveMaxLifetime: frame.lifetime,
-                keepAliveInterval: frame.keepAlive,
-                resumeToken: frame.resumeToken,
-              },
-              requester
-            );
-            const keepAliveHandler = new KeepAliveHandler(
-              connection,
-              frame.lifetime
+            const multiplexerDumiltiplexer = new ResumableClientServerInputMultiplexerDemultiplexer(
+              StreamIdGenerator.create(0),
+              outbound,
+              outbound,
+              new FrameStore(), // TODO: add size parameter
+              frame.resumeToken.toString(),
+              this.sessionStore,
+              this.sessionTimeout
             );
-            const keepAliveSender = this.serverSideKeepAlive
-              ? new KeepAliveSender(
-                  connection.connectionOutbound,
-                  frame.keepAlive
-                )
-              : undefined;
-            const connectionFrameHandler = new ConnectionFrameHandler(
-              connection,
-              keepAliveHandler,
-              leaseHandler,
-              responder
-            );
-            const streamsHandler = new StreamHandler(responder, 0);
 
-            connection.onClose((e) => {
-              keepAliveSender?.close();
-              keepAliveHandler.close();
-              connectionFrameHandler.close(e);
-            });
-            connection.connectionInbound(
-              connectionFrameHandler.handle.bind(connectionFrameHandler)
-            );
-            connection.handleRequestStream(
-              streamsHandler.handle.bind(streamsHandler)
-            );
+            this.sessionStore[
+              frame.resumeToken.toString()
+            ] = multiplexerDumiltiplexer;
 
-            keepAliveHandler.start();
-            keepAliveSender?.start();
-          } catch (e) {
-            connection.close(
-              e instanceof RSocketError
-                ? e
-                : new RSocketError(ErrorCodes.REJECTED_SETUP, e.message)
-            );
+            return multiplexerDumiltiplexer;
           }
-          return;
-        }
-        default: {
-          connection.close(new RSocketError(ErrorCodes.UNSUPPORTED_SETUP));
         }
+
+        return new ClientServerInputMultiplexerDemultiplexer(
+          StreamIdGenerator.create(0),
+          outbound,
+          outbound
```

Comment:
Passing `outbound` twice seems really strange, but I get that we want to access its closeable properties.

Reply:
yeah. we need to figure out better composition. I ended up that Outbound passed to multiplexer has to be decomposed since for resume support the delegate Multiplexer & Demultuplxere has to be unaware of transport disconnections. Thus we pass just `new Deferred()` in the constructor of `ResumableClientServerMultiplexerDemultiplexer`